src/Sulu/Bundle/AdminBundle/Resources/js/containers/Badge/stores/BadgeStore.js

```diff
@@ -0,0 +1,93 @@
+// @flow
+import {observable, action, autorun} from 'mobx';
+import jexl from 'jexl';
+import jsonPointer from 'json-pointer';
+import symfonyRouting from 'fos-jsrouting/router';
+import Router from '../../../services/Router';
+import Requester from '../../../services/Requester';
+
+export default class BadgeStore {
+    router: Router;
+    routeName: string;
+    dataPath: ?string;
+    visibleCondition: ?string;
+    attributesToRequest: Object;
+    routerAttributesToRequest: Object;
+    @observable text: ?string = null;
+    disposer: () => {};
+
+    constructor(
+        router: Router,
+        routeName: string,
+        dataPath: ?string,
+        visibleCondition: ?string,
+        attributesToRequest: Object,
+        routerAttributesToRequest: Object
+    ) {
+        this.router = router;
+        this.routeName = routeName;
+        this.dataPath = dataPath;
+        this.visibleCondition = visibleCondition;
+        this.attributesToRequest = attributesToRequest;
+        this.routerAttributesToRequest = routerAttributesToRequest;
+
+        this.disposer = autorun(() => {
+            this.load();
+        });
+    }
+
+    load = () => {
+        const {
+            router: {
+                attributes: routerAttributes,
+            },
+            routeName,
+            attributesToRequest,
+            routerAttributesToRequest,
+        } = this;
+
+        let requestAttributes = {};
```

Comment:
i think we evaluated this inside of the component in other places right? not sure if there was a specific reason for that